samples/durable-task-sdks/dotnet/Agents/PromptChaining/Worker/Activities/ContentActivities.cs

```diff
@@ -0,0 +1,360 @@
+                    var image = input.images[1];
+                    htmlBuilder.AppendLine("<div class=\"image-container\">");
+                    htmlBuilder.AppendLine($"    <img src=\"{image.ImageUrl}\" alt=\"{image.Description}\">");
+                    htmlBuilder.AppendLine($"    <div class=\"caption\">{image.Caption}</div>");
+                    htmlBuilder.AppendLine("</div>");
+                }
+                
+                // Add the last paragraph
+                if (paragraphs.Count > 1)
+                {
+                    htmlBuilder.AppendLine(paragraphs[paragraphs.Count - 1]);
+                }
+                
+                // Add any remaining images
+                for (int i = 2; i < input.images.Count; i++)
+                {
+                    var image = input.images[i];
+                    htmlBuilder.AppendLine("<div class=\"image-container\">");
+                    htmlBuilder.AppendLine($"    <img src=\"{image.ImageUrl}\" alt=\"{image.Description}\">");
+                    htmlBuilder.AppendLine($"    <div class=\"caption\">{image.Caption}</div>");
+                    htmlBuilder.AppendLine("</div>");
+                }
+            }
+            else
+            {
+                // If we couldn't extract paragraphs, use the original content
+                htmlBuilder.AppendLine(htmlContent);
+                
+                // Add images at the end
+                foreach (var image in input.images)
+                {
+                    htmlBuilder.AppendLine("<div class=\"image-container\">");
+                    htmlBuilder.AppendLine($"    <img src=\"{image.ImageUrl}\" alt=\"{image.Description}\">");
+                    htmlBuilder.AppendLine($"    <div class=\"caption\">{image.Caption}</div>");
+                    htmlBuilder.AppendLine("</div>");
+                }
+            }
+            
+            // Add metadata at the bottom
+            htmlBuilder.AppendLine("<div class=\"metadata\">");
+            htmlBuilder.AppendLine($"    <p>Generated on: {DateTime.UtcNow:yyyy-MM-dd HH:mm:ss} UTC</p>");
+            htmlBuilder.AppendLine("    <p>Images generated with DALL-E</p>");
+            htmlBuilder.AppendLine("    <p>Research conducted with web search</p>");
+            htmlBuilder.AppendLine("</div>");
+            
+            htmlBuilder.AppendLine("</body>");
+            htmlBuilder.AppendLine("</html>");
+            
+            string finalHtml = htmlBuilder.ToString();
+            
+            // Generate a filename based on title and timestamp
+            string sanitizedTitle = SanitizeForFileName(title);
+            string timestamp = DateTime.UtcNow.ToString("yyyyMMddHHmmss");
+            string filename = $"{sanitizedTitle}-{timestamp}.html";
+            
+            // Create path to project's tmp directory
+            string projectRoot = Path.GetFullPath(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "../../../../"));
```

Comment:
This works when deployed to Azure?  Perhaps consider using the "official" temp folder (e.g. `Path.GetTempPath()`) or make the path configurable?